java/vector/src/main/java/org/apache/arrow/vector/compression/CompressionUtil.java

```diff
@@ -47,14 +57,21 @@ public static ArrowBodyCompression createBodyCompression(CompressionCodec codec)
   }
 
   /**
-   * Creates the {@link CompressionCodec} given the compression type.
+   * Process compression by compressing the buffer as is.
    */
-  public static CompressionCodec createCodec(byte compressionType) {
-    switch (compressionType) {
-      case NoCompressionCodec.COMPRESSION_TYPE:
-        return NoCompressionCodec.INSTANCE;
-      default:
-        throw new IllegalArgumentException("Compression type not supported: " + compressionType);
-    }
+  public static ArrowBuf compressRawBuffer(BufferAllocator allocator, ArrowBuf inputBuffer) {
+    ArrowBuf compressedBuffer = allocator.buffer(SIZE_OF_UNCOMPRESSED_LENGTH + inputBuffer.writerIndex());
+    compressedBuffer.setLong(0, NO_COMPRESSION_LENGTH);
+    compressedBuffer.setBytes(SIZE_OF_UNCOMPRESSED_LENGTH, inputBuffer, 0, inputBuffer.writerIndex());
+    compressedBuffer.writerIndex(SIZE_OF_UNCOMPRESSED_LENGTH + inputBuffer.writerIndex());
+    return compressedBuffer;
+  }
+
+  /**
+   * Process decompression by decompressing the buffer as is.
```

Comment:
something like `transferUncompressedComponent` or `extractUncompressedBuffer`

Reply:
I chose `extractUncompressedBuffer`. Thanks for the good suggestion.

Reply:
please update the docs to match, something like.

"Slice the buffer to contain the uncompressed bytes"